lib/livebook/ecto_types/secret_origin.ex

```diff
@@ -1,45 +1,54 @@
 defmodule Livebook.EctoTypes.SecretOrigin do
   @moduledoc false
+
   use Ecto.Type
 
   @type t :: nil | :session | :startup | :app | {:hub, String.t()}
 
+  @impl true
   def type, do: :string
 
-  def load("session"), do: {:ok, :session}
-  def load("app"), do: {:ok, :app}
-  def load("startup"), do: {:ok, :startup}
-
-  def load("hub-" <> id) do
-    if hub_secret?(id),
-      do: {:ok, {:hub, id}},
-      else: :error
-  end
-
-  def load(_), do: :error
+  @impl true
+  def load(origin), do: decode(origin)
 
+  @impl true
   def dump(:session), do: {:ok, "session"}
   def dump(:app), do: {:ok, "app"}
   def dump(:startup), do: {:ok, "startup"}
-
-  def dump({:hub, id}) when is_binary(id) do
-    if hub_secret?(id), do: {:ok, "hub-#{id}"}, else: :error
-  end
```

Comment:
@aleDsz if non-existence is ever a concern we should handle it elsewhere, I don't think it should impact how the type loading/dumping behaves.

Reply:
I agree with you